### PR TITLE
Add a configuration for `additional_columns` at the Resource level

### PR DESCRIPTION
**Pull request by jljaco**
Issue #, if available:  [1281](https://github.com/aws-controllers-k8s/community/issues/1281)

Description of changes:

We want to have the option to include additional printer columns in the output of `kubectl`.  This change allow the author to specify an arbitrary number of those per Resource, for each specifying the name, type, and JSON path of the field to produce.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/config/resource.go [hunk 1]
```diff
@@ -317,6 +317,12 @@ type UpdateOperationConfig struct {
 	CustomMethodName string `json:"custom_method_name"`
 }
 
+type AdditionalColumnConfig struct {
+	Name string `json:"name"`
+	JSONPath string `json:"json_path"`
+	Type string `json:"type"`
+}
```
Comment (A-Hilaly) [inline pkg/config/resource.go:324]: nit: Please include tiny comments for the struct and its fields. Those will be included in our generated GoDoc documentation

File: pkg/model/model.go [hunk 2]
```diff
@@ -284,6 +284,16 @@ func (m *Model) GetCRDs() ([]*CRD, error) {
 			crd.AddStatusField(memberNames, memberShapeRef)
 		}
 
+		// Now add the additional printer columns that have been defined explicitly
+		// in additional_columns
+		for _, additionalColumn := range m.cfg.GetAdditionalColumns(crdName) {
```
[the remark this answers is not in the document]
Comment (jljaco) [reply pkg/model/model.go]: Done

File: pkg/config/resource.go (hunk 1 above)
Comment (jljaco) [reply pkg/config/resource.go]: Done

File: pkg/model/model.go [hunk 3]
```diff
@@ -284,6 +284,16 @@ func (m *Model) GetCRDs() ([]*CRD, error) {
...
+			printerColumn := &PrinterColumn{}
+			printerColumn.Name = additionalColumn.Name
+			printerColumn.JSONPath = additionalColumn.JSONPath
+			printerColumn.Type = additionalColumn.Type
```
[the remark this answers is not in the document]
Comment (jljaco) [reply pkg/model/model.go]: Done

File: pkg/config/resource.go [hunk 4]
```diff
@@ -334,6 +340,10 @@ type PrintConfig struct {
 	AddSyncedColumn *bool `json:"add_synced_column"`
 	// OrderBy is the field used to sort the list of PrinterColumn options.
 	OrderBy string `json:"order_by"`
+
+	// AdditionalColumns can be used to add arbitrary extra columns to a Resource's output
+	// if present, should be a list of objects, each containing: name, json_path, and type
+	AdditionalColumns []AdditionalColumnConfig `json:"additional_columns,omitempty"`
```
[the remark this answers is not in the document]
Comment (jljaco) [reply pkg/config/resource.go]: Done

File: pkg/config/config.go [hunk 5]
```diff
@@ -75,14 +75,14 @@ type PrefixConfig struct {
 }
 
 // GetAdditionalColumns extracts AdditionalColumns defined for a given Resource
-func (c *Config) GetAdditionalColumns(resourceName string) []AdditionalColumnConfig {
+func (c *Config) GetAdditionalColumns(resourceName string) []*AdditionalColumnConfig {
 	if c == nil {
-		return []AdditionalColumnConfig{}
+		return []*AdditionalColumnConfig{}
```
Comment (A-Hilaly) [inline pkg/config/config.go:80]: Just a nil return is sufficient
```suggestion
        return nil
```

File: pkg/config/config.go [hunk 6]
```diff
@@ -75,14 +75,14 @@ type PrefixConfig struct {
...
 	}
 
 	resourceConfig, ok := c.Resources[resourceName]
-	if !ok || resourceConfig.Print == nil || resourceConfig.Print.AdditionalColumns == nil {
-		return []AdditionalColumnConfig{}
+	if !ok || resourceConfig.Print == nil || len(resourceConfig.Print.AdditionalColumns) == 0 {
+		return []*AdditionalColumnConfig{}
```
Comment (A-Hilaly) [inline pkg/config/config.go:85]: Ditto

File: pkg/config/resource.go [hunk 7]
```diff
@@ -317,10 +317,20 @@ type UpdateOperationConfig struct {
...
 	Name string `json:"name"`
+	// the JSONPath definining the source of the output
 	JSONPath string `json:"json_path"`
+	// the OpenAPI type of the output
+	// c.f., https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types
 	Type string `json:"type"`
+	// the priority of the column in the resource's output
+	Priority int `json:"priority,omitempty"`
+	// the zero-based index of the position at which to display the column in output
+	Index int `json:"index,omitempty"`
```
Comment (A-Hilaly) [inline pkg/config/resource.go:333]: Quotting from https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences:
> Comments should begin with the name of the thing being described and end in a period

File: pkg/model/crd.go [hunk 8]
```diff
@@ -566,6 +566,20 @@ func (r *CRD) PrintSyncedColumn() bool {
...
+		printerColumn := &PrinterColumn{}
+		printerColumn.Name = additionalColumn.Name
+		printerColumn.JSONPath = additionalColumn.JSONPath
+		printerColumn.Type = additionalColumn.Type
+		printerColumn.Priority = additionalColumn.Priority
+		printerColumn.Index = additionalColumn.Index
+		r.additionalPrinterColumns = append(r.additionalPrinterColumns, printerColumn)
+	}
+}
+
```
Comment (A-Hilaly) [inline pkg/model/crd.go:582]: The allocation on L570 overrides the existing printerColumns previously added in `model.go` L284 and L236. Removing L570 should fix the code and the unit tests

File: pkg/config/config.go (hunk 6 above)
Comment (jljaco) [reply pkg/config/config.go]: Done

File: pkg/config/resource.go (hunk 7 above)
Comment (jljaco) [reply pkg/config/resource.go]: Done

**Review by A-Hilaly — APPROVED** [review]
Top! Merci beaucoup :)
/lgtm

**Pull request merged**